build-tools/packages/build-tools/src/repoPolicyCheck/handlers/pnpm.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import fs from "fs";
+import path from "path";
+
+import { PackageJson } from "../../common/npmPackage";
+import { Handler, readFile } from "../common";
+
+const match = /(?:^|\/)pnpm-lock\.yaml$/i;
+export const handlers: Handler[] = [
+	{
+		name: "pnpm-npm-package-json-preinstall",
+		match,
+		handler: (file) => {
+			const dirname = path.dirname(file);
+			const packageJsonFile = path.join(dirname, "package.json");
+			let json: PackageJson;
+			try {
+				json = JSON.parse(readFile(packageJsonFile));
+			} catch (err) {
+				return "Error parsing JSON file: " + packageJsonFile;
+			}
+
+			const script: string | undefined = json.scripts?.preinstall;
+			if (script) {
+				const matchResult = script.match(/^node ((?:\.\.\/)*)scripts\/only-pnpm.cjs/);
```

Comment:
yeah... but at the time to the repo-check runs, the script needs to exists.  And I believe the repo-policy runs on everything in the repo?  So that would be a problem.